src/System.Net.Sockets/tests/FunctionalTests/SelectTest.cs

```diff
@@ -79,6 +79,37 @@ public void Select_ReadWrite_AllReady(int reads, int writes)
             }
         }
 
+        [Theory]
+        [InlineData(2, 0)]
+        [InlineData(2, 1)]
+        [InlineData(2, 2)]
+        [InlineData(2, 3)]
+        [InlineData(2, 4)]
+        [InlineData(2, 5)]
+        public void Select_SocketAlreadyClosed_AllSocketsClosableAfterException(int socketsPerType, int indexToDispose)
+        {
+            KeyValuePair<Socket, Socket>[] socketPairs = Enumerable.Range(0, socketsPerType * 3).Select(_ => CreateConnectedSockets()).ToArray();
+            try
+            {
+                Socket[] reads = socketPairs.Take(socketsPerType).Select(p => p.Key).ToArray();
+                Socket[] writes = socketPairs.Skip(socketsPerType).Take(socketsPerType).Select(p => p.Key).ToArray();
+                Socket[] errors = socketPairs.Skip(socketsPerType * 2).Take(socketsPerType).Select(p => p.Key).ToArray();
+
+                socketPairs[indexToDispose].Key.Dispose();
+
+                Assert.Throws<ObjectDisposedException>(() => Socket.Select(reads, writes, errors, 1_000));
+
+                for (int i = 0; i < socketPairs.Length; i++)
+                {
+                    Assert.Equal(i == indexToDispose, socketPairs[i].Key.SafeHandle.IsClosed);
+                }
+            }
+            finally
+            {
+                DisposeSockets(socketPairs);
```

Comment:
nit: looking at the test method, it's not clear that `DisposeSockets` will Assert `IsClosed`.